test/plug/adapters/cowboy/conn_test.exs

```diff
@@ -129,6 +135,34 @@ defmodule Plug.Adapters.Cowboy.ConnTest do
            {"x-sample", "value"}
   end
 
+  test "allows customized statuses based on config" do
+    assert {500, _headers, _} = request :get, "/send_451"
+
+    Application.put_env(:plug, :statuses, %{451 => "Unavailable For Legal Reasons"})
```

Comment:
> @josevalim there isn't currently a config.exs for plug. Is it worth creating one just for this case?

Yes.

> I like it this way because we can also test what happens when no custom statuses are set.

If you prefer then you can set the custom environment only during tests by wrapping the config in `config/config.exs` in a `Mix.env == :test` check.